### PR TITLE
Introducing `transferFrom` function summary on `dAIMock`

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
This PR adds the summarized rules regarding the program execution. The rule presented in this PR summarizes the function of `transferFrom` function on `dAIMock`.

By summarizing this function, we save 1265 steps.
(Without it, the program took 19055 steps, and after implementing it, the program took 17790 steps)

Wall Time to execute 1000 `testSwapSingleHopExactAmountIn` summarized:
- On `main` branch: `0.176s`
- On this branch: `0.170s`

Wall Time to execute 1000 `testSwapSingleHopExactAmountIn` without any summarization: `0.269s`

### CODE REVIEW

File: src/uniswap-summaries.md [hunk 1]
```diff
@@ -3501,6 +3501,25 @@ module SOLIDITY-UNISWAP-MINT-SUMMARY
...
+  rule <k> transferFrom:Id ( v(V1:MInt{160}, address #as T), v(V2:MInt{160}, T), v(V3:MInt{256}, uint256), .TypedVals ) ~> Ss => v ( true , bool ) ~> Ss </k>
+        <summarize> true </summarize>
+        <this> THIS </this>
+        <contract-address> THIS </contract-address>
+        <store> S => S ListItem(V1) // src
+                       ListItem(V2) // dst
+                       ListItem(V3) // wad
+        </store>
+        <contract-storage> Storage => Storage [ balanceOf <- write({write({Storage [ balanceOf ] orDefault .Map}:>Value, ListItem(V1), ({read({Storage [ balanceOf ] orDefault .Map}:>Value, ListItem(V1), (mapping ( address account => uint256 )))}:>MInt{256} -MInt V3:MInt{256}), (mapping ( address account => uint256))) }:>Value, ListItem(V2), ({read({Storage [ balanceOf ] orDefault .Map}:>Value, ListItem(V2), (mapping ( address account => uint256 )))}:>MInt{256} +MInt V3:MInt{256}), (mapping ( address account => uint256))) ]
+        </contract-storage> [priority(40)]
```
Comment (mariaKt) [inline src/uniswap-summaries.md:3519]: This is definitely not a correct summary. I see no two cases, therefore this is only summarizing only one of the two branches (the if not taken). This is OK, if you did not intend to do both. But, the conditions that make sure this rule is not applicable if the condition is true are missing.
The requirement for the requires statement at the beginning is also missing.

**Review by mariaKt — CHANGES_REQUESTED** [review]
I requested some changes, pertaining to adding side conditions to the rule. Please let me know if I have misunderstood anything.

File: src/uniswap-summaries.md [hunk 2]
```diff
@@ -3535,6 +3535,56 @@ module SOLIDITY-UNISWAP-APPROVE-SUMMARY
...
+```k
+module SOLIDITY-UNISWAP-TRANSFERFROM-SUMMARY
+  imports SOLIDITY-CONFIGURATION
+  imports SOLIDITY-EXPRESSION
+  imports SOLIDITY-UNISWAP-TOKENS
+
+  rule <k> transferFrom:Id ( v(V1:MInt{160}, address #as T), v(V2:MInt{160}, T), v(V3:MInt{256}, uint256), .TypedVals ) ~> Ss => v ( true , bool ) ~> Ss </k>
+        <summarize> true </summarize>
+        <this> THIS </this>
+        <contract-address> THIS </contract-address>
```
Comment (mariaKt) [inline src/uniswap-summaries.md:3547]: You can match `<this-type>` and `<contract-id>` cells. I posted details on slack.

File: src/uniswap-summaries.md [hunk 3]
```diff
@@ -3535,6 +3535,56 @@ module SOLIDITY-UNISWAP-APPROVE-SUMMARY
...
+        </contract-storage>
+    requires {read({Storage [ balanceOf ] orDefault .Map}:>Value, ListItem(V1), (mapping ( address daiact => uint256 )))}:>MInt{256} >=uMInt V3:MInt{256} // balanceOf[src] >= wad
+     andBool (V1 =/=MInt SENDER orBool {read(read({Storage [ allowance ] orDefault .Map}:>Value, ListItem(V1), (mapping ( address daiownr => mapping ( address daispdr => uint256 ) ))), ListItem(SENDER), (mapping ( address daispdr => uint256 )))}:>MInt{256} =/=MInt {Storage[constUINTMAX] orDefault 0p256}:>MInt{256}) // src != msg.sender && allowance[src][msg.sender] != constUINTMAX
```
Comment (mariaKt) [inline src/uniswap-summaries.md:3581]: Shouldn't this `orBool` be `andBool`?

**Review by mariaKt — CHANGES_REQUESTED** [review]

File: src/uniswap-summaries.md (hunk 3 above)
Comment (Robertorosmaninho) [reply src/uniswap-summaries.md:3581]: Right! My bad!

**Review by mariaKt — APPROVED** [review]
Approved. Thanks for addressing the comments.

**Pull request merged**